src/rendering/SceneManager.cc

```diff
@@ -265,6 +265,11 @@ rendering::VisualPtr SceneManager::CreateVisual(Entity _id,
   rendering::VisualPtr visualVis = this->dataPtr->scene->CreateVisual(name);
   visualVis->SetUserData("gazebo-entity", static_cast<int>(_id));
   visualVis->SetUserData("pause-update", static_cast<int>(0));
+
+  auto geometry = _visual.Geom();
+  if (geometry)
+    visualVis->SetUserData("geometry-type", static_cast<int>(geometry->Type()));
```

Comment:
How about adding a function to `ign-rendering` that returns this information? I think this can only be done from Fortress, so we'll need to keep using user data here.

Reply:
Sounds good to me.